desc/objectives/_omnigenity.py

```diff
@@ -540,6 +541,319 @@ def compute(self, params, constants=None):
+    weight : float, ndarray, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        len(weight) must be equal to Objective.dim_f
+    normalize : bool
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool
+       Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    eq_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at for equilibrium data.
+        Must be a single flux suface.
+    field_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at for omnigenous field data.
+        Must be a single flux suface.
+    M_booz : int, optional
+        Poloidal resolution of Boozer transformation. Default = 2 * eq.M.
+    N_booz : int, optional
+        Toroidal resolution of Boozer transformation. Default = 2 * eq.N.
+    well_weight : float, optional
+        Weight applied to the bottom of the magnetic well (B_min) relative to the top
+        of the magnetic well (B_max). Default = 1, which weights all points equally.
+    eq_fixed: bool, optional
+        Whether the Equilibrium `eq` is fixed or not.
+        If True, the equilibrium is fixed and its values are precomputed, which saves on
+        computation time during optimization and self.things = [field] only.
+        If False, the equilibrium values are computed at every iteration (Default).
+    field_fixed: bool, optional
+        Whether the OmnigenousField `field` is fixed or not.
+        If True, the field is fixed and its values are precomputed, which saves on
+        computation time during optimization and self.things = [eq] only.
+        If False, the field values are computed at every iteration (Default).
+    name : str
+        Name of the objective function.
+
+    """
+
+    _coordinates = "rtz"
+    _units = "(T)"
+    _print_value_fmt = "Omnigenity error: {:10.3e} "
+
+    def __init__(
+        self,
+        eq,
+        field,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="fwd",  # FIXME: get it working with rev mode, set default to auto
```

Comment:
wwhat is  the probelem with reverse mode right now?

Reply:
I suspect it's something related to the transformation from (alpha,eta) to (theta_B, zeta_B) where the matrix contains stuff like 1/N.


This gives a zero division error for me:
```python
eq = desc.examples.get("DSHAPE")
field = desc.magnetic_fields.OmnigenousField(2,2,2,2,2)
obj = Omnigenity(eq, field, deriv_mode="rev")
obj.build()
obj.jac_scaled(eq.params_dict, field.params_dict)
```
Curiously, it doesn't happen if I wrap things in an `ObjectiveFunction`, though it does happen regardless of forward or reverse mode.

Can easily be fixed using the "double where" trick https://jax.readthedocs.io/en/latest/faq.html#gradients-contain-nan-where-using-where

Or, in this case maybe a quadruple where since there are a few different terms that can be 1/0

Reply:
@ddudt should make an issue to fix this in the near term